numpy/_core/src/multiarray/item_selection.c

```diff
@@ -3154,3 +3058,166 @@ PyArray_MultiIndexSetItem(PyArrayObject *self, const npy_intp *multi_index,
 
     return PyArray_Pack(PyArray_DESCR(self), data, obj);
 }
+
+
+/* Table of generic sort functions for use in PyArray_SortEx*/
+static PyArray_SortFunc* const generic_sort_table[] = {npy_quicksort,
+                                                       npy_heapsort,
+                                                       npy_timsort};
+
+/*NUMPY_API
+ * Sort an array in-place with extended parameters
+ */
+NPY_NO_EXPORT int
+PyArray_SortEx(PyArrayObject *op, int axis, NPY_SORTFLAGS flags)
+{
+    PyArray_SortFunc **sort_table = NULL;
+    PyArray_SortFunc *sort = NULL;
+
+    if (check_and_adjust_axis(&axis, PyArray_NDIM(op)) < 0) {
+        return -1;
+    }
+
+    if (PyArray_FailUnlessWriteable(op, "sort array") < 0) {
+        return -1;
+    }
+
+    sort_table = PyDataType_GetArrFuncs(PyArray_DESCR(op))->sort;
+    switch (flags) {
+        case NPY_SORT_STABLE:
+            sort = sort_table[NPY_STABLESORT];
+            break;
+        default:
```

Comment:
I guess this really should be `case 0:` and the default should just fall through.

Reply:
I figured that the compiler would take care of it. Case statements are reconfigured as binary search under the covers.  But I can see your point.

Reply:
Hmm, I never even thought of the compiler for `case NPY_SORT_STABLE` doing the equivalent of `(flags & NPY_SORT_STABLE) == NPY_SORT_STABLE`... that's quite a bit of magic...

Reply:
Should be Fixed.